.changelog/16319.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+api: Fix `Allocations().Stop()` method to properly set the request `LastIndex` and `RequestTime` in the response
```

Comment:
This is the only user facing change, but should I note the future breaking changes somehow?

Reply:
Ah nice. I added the deprecation notes and I also remembered that I added `RestartAllTasks()` method to [prevent a breaking change](https://github.com/hashicorp/nomad/pull/14127#discussion_r948577573). Since we're breaking things already we may as well clean that up.